awslogs/bin.py

```diff
@@ -1,18 +1,62 @@
 import os
+import re
 import sys
 import locale
 import codecs
 import argparse
+from datetime import datetime, timedelta
+from dateutil.parser import parse
+from dateutil.tz import tzutc
 
 import boto3
 from botocore.client import ClientError
+from botocore.compat import total_seconds
 from termcolor import colored
 
 from . import exceptions
 from .core import AWSLogs
 from ._version import __version__
 
 
+def regex_str(s):
+    """Verifies that the s is a valid python regex
+    if s is not a valid regex then an exception is raised"""
+    try:
+        re.compile(s)
+    except Exception as e:
+        raise exceptions.InvalidPythonRegularExpressionError('Log stream name pattern', s)
+
+    return s
```

Comment:
Usually the return value/what is passed around after handling the CLI would be the compiled regex - it may also speed up the code if it the code using it is called more than once, as the cache use isn't free.